lib/ecto/adapters/myxql.ex

```diff
@@ -312,27 +312,42 @@ defmodule Ecto.Adapters.MyXQL do
   def structure_dump(default, config) do
     table = config[:migration_source] || "schema_migrations"
     path  = config[:dump_path] || Path.join(default, "structure.sql")
+    dump_prefixes = config[:dump_prefixes] || [config[:database]]
 
-    with {:ok, versions} <- select_versions(table, config),
-         {:ok, contents} <- mysql_dump(config),
+    with {:ok, versions} <- select_versions(dump_prefixes, table, config),
+         {:ok, contents} <- mysql_dump(dump_prefixes, config),
          {:ok, contents} <- append_versions(table, versions, contents) do
       File.mkdir_p!(Path.dirname(path))
       File.write!(path, contents)
       {:ok, path}
     end
   end
 
-  defp select_versions(table, config) do
-    case run_query(~s[SELECT version FROM `#{table}` ORDER BY version], config) do
-      {:ok, %{rows: rows}} -> {:ok, Enum.map(rows, &hd/1)}
-      {:error, %{mysql: %{name: :ER_NO_SUCH_TABLE}}} -> {:ok, []}
+  defp select_versions(dump_prefixes, table, config) do
+    result = Enum.reduce_while(dump_prefixes, [], fn prefix, versions ->
+      case run_query(~s[SELECT version FROM `#{prefix}`.`#{table}` ORDER BY version], config) do
+        {:ok, %{rows: rows}} -> {:cont, Enum.map(rows, &{prefix, hd(&1)}) ++ versions }
+        {:error, %{mysql: %{name: :ER_NO_SUCH_TABLE}}} -> {:cont, versions}
+        {:error, _} = error -> {:halt, error}
+        {:exit, exit} -> {:error, exit_to_exception(exit)}
+      end
+    end)
```

Comment:
```suggestion
  defp select_versions(prefixes, table, config) do
    result =
      Enum.reduce_while(prefixes, [], fn prefix, versions ->
        case run_query(~s[SELECT version FROM `#{prefix}`.`#{table}` ORDER BY version], config) do
          {:ok, %{rows: rows}} -> {:cont, Enum.map(rows, &{prefix, hd(&1)}) ++ versions}
          {:error, %{mysql: %{name: :ER_NO_SUCH_TABLE}}} -> {:cont, versions}
          {:error, _} = error -> {:halt, error}
          {:exit, exit} -> {:halt, {:error, exit_to_exception(exit)}}
        end
      end)
```
- formatting
- halting on `:exit`
- calling it `prefixes` instead of `dump_prefixes` to be consistent with postgres and also because we are inside the dump functions